MatrixSDK/Crypto/MXCrypto.m

```diff
@@ -2880,6 +2909,35 @@ - (MXHTTPOperation *)generateAndUploadOneTimeKeys:(NSUInteger)keyCount retry:(BO
     return operation;
 }
 
+- (MXHTTPOperation *)generateAndUploadFallbackKey
+{
+    [_olmDevice generateFallbackKey];
+    
+    NSDictionary *fallbackKey = _olmDevice.fallbackKey;
+    NSMutableDictionary *fallbackKeyJson = [NSMutableDictionary dictionary];
+    
+    for (NSString *keyId in fallbackKey[kMXKeyCurve25519Type])
+    {
+        // Sign the fallback key
+        NSMutableDictionary *signedKey = [NSMutableDictionary dictionary];
+        signedKey[@"key"] = fallbackKey[kMXKeyCurve25519Type][keyId];
+        signedKey[@"signatures"] = [self signObject:signedKey];
+        
+        fallbackKeyJson[[NSString stringWithFormat:@"%@:%@", kMXKeySignedCurve25519Type, keyId]] = signedKey;
+    }
+    
+    MXLogDebug(@"[MXCrypto] generateAndUploadFallbackKey: Started uploading fallback key.");
+    
+    MXWeakify(self);
+    return [_matrixRestClient uploadKeys:nil oneTimeKeys:nil fallbackKeys:fallbackKeyJson success:^(MXKeysUploadResponse *keysUploadResponse) {
+        weakself.uploadFallbackKeyOperation = nil;
```

Comment:
We never use `weakself` directly but `MXStrongifyAndReturnIfNil(self);` instead.
This method logs if the weak ref has been released. This is useful when analysing rageshake logs.

Reply:
Right, got you! 👍